misk/testing/src/main/kotlin/misk/testing/MiskTestExtension.kt

```diff
@@ -56,6 +61,19 @@ internal class MiskTestExtension : BeforeEachCallback, AfterEachCallback {
     }
   }
 
+  /** We inject after starting services and uninject after stopping services. */
+  @Singleton
+  class InjectUninject : BeforeEachCallback, AfterEachCallback {
+    override fun beforeEach(context: ExtensionContext) {
+      val injector = context.retrieve<Injector>("injector")
+      injector.injectMembers(context.requiredTestInstance)
+    }
+
+    override fun afterEach(context: ExtensionContext) {
+      uninject(context.requiredTestInstance)
```

Comment:
OOC why do we uninject?

Reply:
Mostly because JUnit keeps instances of the test class in memory for the duration of the test run. If the test has nontrivial things injected, that ends up eating all the memory and slowing down a long test suite.